packages/remix-react/data.ts

```diff
@@ -54,11 +57,32 @@ export async function fetchData(
   return response;
 }
 
-export async function extractData(response: Response): Promise<AppData> {
+export async function extractData(
+  response: Response,
+  signal: AbortSignal,
+  resolveDeferred = false
+): Promise<unknown | DeferredData> {
   // This same algorithm is used on the server to interpret load
   // results when we render the HTML page.
   let contentType = response.headers.get("Content-Type");
 
+  if (
+    response.body &&
+    contentType &&
+    /\btext\/remix-deferred\b/.test(contentType)
+  ) {
+    let deferred = await parseDeferredReadableStream(response.body);
+    if (!resolveDeferred) {
+      return deferred;
+    }
+
+    if (await deferred.resolveData(signal)) {
```

Comment:
Would you mind adding a comment here that explains what this is doing?

Reply:
I want to review this behavior more. It's and edge case so we might just want this to throw but I'd have to look deeper.